code/game/objects/items.dm

```diff
@@ -846,17 +846,13 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 		// Create a callback with checks that would be called every tick by do_after.
 		var/datum/callback/tool_check = CALLBACK(src, .proc/tool_check_callback, user, amount, extra_checks)
 
-		if(ismob(target))
-			if(!do_mob(user, target, delay, extra_checks=tool_check))
-				return
+		if(ismob(target) && !do_mob(user, target, delay, extra_checks=tool_check))
+			return
 
-		else
-			if(!do_after(user, delay, target=target, extra_checks=tool_check))
-				return
-	else
-		// Invoke the extra checks once, just in case.
-		if(extra_checks && !extra_checks.Invoke())
+		else if(!do_after(user, delay, target=target, extra_checks=tool_check))
```

Comment:
This logic is problematic. This `else` will trigger for both when the target is not a mob (as should be) but also if it fails the `do_mob` (that is not intended).
Keep the old indentation on the `ismob` and `do_mob` checks.

Reply:
I have fixed those issues now